enzy_htp/structure/structure_region/capping.py

```diff
@@ -50,124 +62,222 @@ def capping_with_residue_terminals(raw_region,
                       " composed by whole residues only.")
         raise ValueError
 
-    free_res_mapper = raw_region.involved_residues_with_free_terminal()
-    free_nter_res = free_res_mapper["n_ter"]
-    free_cter_res = free_res_mapper["c_ter"]
-    
-    add_atoms = []
-    for res in free_nter_res:
-        cap_atoms = cap_residue_free_terminal(res, nterm_cap, "nterm")
-        add_atoms.extend(cap_atoms)
-
-    for res in free_cter_res:
-        cap_atoms = cap_residue_free_terminal(res, cterm_cap, "cterm")
-        add_atoms.extend(cap_atoms)
-    
+    new_atoms:List[Atom] = list()
+    for res in raw_region.involved_residues:
+        
+        if raw_region.needs_nterm_cap(res):
+            new_atoms.extend(cap_residue_free_terminal(res, nterm_cap, "nterm"))
+
+        new_atoms.extend(res.atoms)
+
+        if raw_region.needs_cterm_cap(res):
+            new_atoms.extend(cap_residue_free_terminal(res, cterm_cap, "cterm"))
+
     if return_copy:
         new_region = raw_region.clone()
-        new_region.atoms.extend(add_atoms)
+        new_region.atoms = new_atoms 
         return new_region
     else:
-        raw_region.atoms.extend(add_atoms)
-
-# endregion
+        raw_region.atoms = new_atoms 
 
-class ResidueCap(Residue):
-    """the residue cap that link to the corresponding
+class ResidueCap(Residue, ABC):
+    """The residue cap that link to the corresponding
     Residue while not having the children atoms actually in
-    that Residue.
+    that Residue. It ensures that given the defined terminal atom, it can 
+    generate a realistic geometry.
+
     This is mainly used in handling cases like capping atoms that
-    does not exists in the real structure.
-    Attribute:
-        link_residue
-        children/atoms
-        parent = None
-        link_atom:
-            the linking atom in original res
-        socket_atom:
-            the socket from original stru to pluging the cap.
-            the cap is replacing the atom.
-        plug_atom:
-            the pluging in atom in the cap
+    do not exist in the real structure.
+
+    Attributes:
+        link_residue: The Residue() being capped.
+        link_atom: THe terminal Atom() that needs capping specifically.
```

Comment:
911 I report a typo here

Reply:
Whats the name of the typo? What's the address?

Reply:
I removed THe typo.